test/e2e/framework/expect.go

```diff
@@ -43,7 +46,22 @@ func ExpectNoError(err error, explain ...interface{}) {
 // ExpectNoErrorWithOffset checks if "err" is set, and if so, fails assertion while logging the error at "offset" levels above its caller
 // (for example, for call chain f -> g -> ExpectNoErrorWithOffset(1, ...) error would be logged for "f").
 func ExpectNoErrorWithOffset(offset int, err error, explain ...interface{}) {
-	gomega.ExpectWithOffset(1+offset, err).NotTo(gomega.HaveOccurred(), explain...)
+	if err == nil {
+		return
+	}
+
+	// Errors usually contain unexported fields. We have to use
+	// a formatter here which can print those.
+	Logf("Unexpected error:\n%s", format.Object(err, 1))
```

Comment:
Consider including the explain text in this. Something like this?

```suggestion
	Logf("Unexpected error: %s\n%s", prefix, format.Object(err, 1))
```

Reply:
Makes sense, added.